### PR TITLE
chore: Use Secure Storage Connector Module in BYOB example

### PR DESCRIPTION
**Pull request by elainaRenee**

### CODE REVIEW

File: examples/byob/main.tf [hunk 1]
```diff
@@ -30,98 +30,17 @@ locals {
...
 }
 
-data "aws_caller_identity" "current" {}
-
-resource "aws_kms_key" "key" {
-  key_usage   = "ENCRYPT_DECRYPT"
-  description = "Managed key to encrypt and decrypt storage file"
-}
-
-resource "aws_kms_key_policy" "key_policy" {
```
Comment (elainaRenee) [inline examples/byob/main.tf:40]: Btw, this resource came out in a terraform version greater than what we currently support for our AWS modules

Comment (venky-wandb) [reply examples/byob/main.tf:40]: Oh, it was always there and we never ran into an issue? 🤔

**Review by venky-wandb — APPROVED** [review]
I think the secure storage connector has already been merged so i haven't looked at that part.  As of this PR, it looks good! 👍

File: examples/byob/main.tf (hunk 1 above)
Comment (elainaRenee) [reply examples/byob/main.tf:40]: We changed the example to use this resource semi-recently. I think because this is just an example it went unnoticed!

**Pull request merged**